git_machete/client.py

```diff
@@ -1682,18 +1682,21 @@ def checkout_github_prs(self, pr_no: int) -> None:
         remote: str
         remote, (org, repo) = self.__derive_remote_and_github_org_and_repo()
         debug('checkout_github_pr()', f'organization is {org}, repository is {repo}')
+        print(f"Fetching {remote}...")
+        self.__git.fetch_remote(remote)
+        if self.__git.remotes():
+            self.flush_caches()
 
-        all_prs: List[GitHubPullRequest] = derive_pull_requests(org, repo)
-        pr = utils.find_or_none(lambda pr: pr.number == pr_no, all_prs)
-
+        pr = get_pull_request_by_number_or_none(str(pr_no), org, repo)
```

Comment:
I'd rather expect the `get_pull_request_by_number_or_none`'s first param to be an `int` as well (since you're expecting a PR number here... not just any string), the conversion to `str` is an implementation detail